itou/www/gps/views.py

```diff
@@ -239,3 +238,230 @@ def display_contact_info(request, group_id, target_participant_public_id, mode):
     )
     log_contact_info_display(request.user, follow_up_group, target_participant, mode)
     return render(request, template_name, {"member": target_participant})
+
+
+@check_user(is_allowed_to_use_gps)
+def join_group(request, template_name="gps/join_group.html"):
+    urls = {
+        Channel.FROM_COLLEGUE: reverse("gps:join_group_from_collegue"),
+        Channel.FROM_NIR: reverse("gps:join_group_from_nir"),
+        Channel.FROM_NAME_EMAIL: reverse("gps:join_group_from_name_and_email"),
+    }
+    if request.current_organization is None:
+        return HttpResponseRedirect(urls[Channel.FROM_NAME_EMAIL])
+
+    form = JoinGroupChannelForm(data=request.POST or None)
+    if request.POST and form.is_valid():
+        return HttpResponseRedirect(urls[form.cleaned_data["channel"]])
+
+    context = {
+        "back_url": get_safe_url(request, "back_url", fallback_url=reverse_lazy("gps:group_list")),
+        "can_use_gps_advanced_features": is_allowed_to_use_gps_advanced_features(request.user),
+    }
+    return render(request, template_name, context)
+
+
+@check_user(is_allowed_to_use_gps)
+def join_group_from_collegue(request, template_name="gps/join_group_from_collegue.html"):
```

Comment:
```suggestion
def join_group_from_colleague(request, template_name="gps/join_group_from_colleague.html"):
```
(et surement plein d'autres :sweat_smile: )